### PR TITLE
Added permalink time variables

### PR DESCRIPTION
**Pull request by pheuko**
This is the solution that I thought for issue #3978. 

You just need to configure your permalink with the new timestamp variable and put the time in the document's filename.

You can also override the date via the document's YAML front matter but don't forget about your environment timezone.

Is this what you were talking about?

### CODE REVIEW

**Comment by envygeeks** [comment]
I'm just gonna say this is one of the most complete pull requests I've seen in a long time.  :+1: and :shipit:

**Comment by pheuko** [comment]
Thank you @envygeeks

**Comment by parkr** [comment]
Nice work, @pheuko! People might want to customize this, though, which I would prefer we avoid. Would it be better to just add the three levels of time and support that?

**Comment by jaybe-jekyll** [comment]
This is nice. Thanks for disabling the regex capture of the added valid post name regex. `?:` :)

**Comment by pheuko** [comment]
Yes, custom time will be better. Hour, minute and second, right? I will change to this.

**Comment by pheuko** [comment]
@jaybe-jekyll :+1:

**Comment by parkr** [comment]
> Yes, custom time will be better. Hour, minute and second, right? I will change to this.

Yes! `:hour`, `:minute`, `:seconds` and the corresponding tests & docs. You da :bomb:, thanks!

**Comment by pheuko** [comment]
I just updated my PR with time variables and created an example about the timestamp permalink. Surely is much more customizable, please check if it is ok.

File: lib/jekyll/post.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class Post
     include Convertible
 
     # Valid post name regex.
-    MATCHER = /^(.+\/)*(\d+-\d+-\d+)-(.*)(\.[^.]+)$/
+    MATCHER = /^(.+\/)*(\d+-\d+-\d+(?:.\d+:\d+:\d+)?)-(.*)(\.[^.]+)$/
```
Comment (parkr) [inline lib/jekyll/post.rb]: I don't think this is a good plan – this should be specified in the `date` front matter attribute. Please revert this and use the calculated date from the post's front matter.

**Comment by parkr** [comment]
Change to the post filename is out of scope but everything else looks good. Please revert the `MATCHER` change and related docs/tests.

Thanks!

**Comment by pheuko** [comment]
Ok

**Comment by pheuko** [comment]
Reverting the MATCHER this PR becomes really simple. =)

**Comment by parkr** [comment]
:shipit:

**Comment by envygeeks** [comment]
:heart:

**Comment by Fastidious** [comment]
Was this implemented? Running `jekyll 3.0.0.pre.beta9` and I am getting literal `:hour:minute` on the permalink, when using:

`permalink: /:year:month:day:hour:minute`

**Comment by parkr** [comment]
@Fastidious It's in `master` – i don't believe it made it to beta98.

**Comment by Fastidious** [comment]
@parkr a `gem install jekyll` Is giving me 2.5.3, how do I go about getting master with gem?

**Comment by envygeeks** [comment]
@Fastidious you can't gem install master, you either need to use our master docker image or you need to use a Gemfile and tell it to pull github source and install it.

**Comment by Fastidious** [comment]
@envygeeks how long it takes for the changes in master to make it to RubyGems?

**Comment by envygeeks** [comment]
I'll ping @parkr and ask him to release.

**Comment by Fastidious** [comment]
@envygeeks was @parkr pinged?

**Comment by parkr** [comment]
@Fastidious https://github.com/jekyll/jekyll/releases/tag/v3.0.0.pre.beta10

**Comment by Fastidious** [comment]
@parkr, thanks!

**Pull request merged**